modules/gapi/src/streaming/onevpl/accelerators/accel_policy_dx11.cpp

```diff
@@ -0,0 +1,117 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2021 Intel Corporation
+
+#ifdef HAVE_ONEVPL
+#include "streaming/onevpl/accelerators/accel_policy_dx11.hpp"
+//#include "streaming/vpl/vpl_utils.hpp"
+#include "streaming/onevpl/accelerators/surface/cpu_frame_adapter.hpp"
+#include "streaming/onevpl/accelerators/surface/surface.hpp"
+#include "logger.hpp"
+
+#ifdef HAVE_DIRECTX
+    #ifdef HAVE_D3D11
+        #pragma comment(lib,"d3d11.lib")
+
+        #define D3D11_NO_HELPERS
+        #include <d3d11.h>
+        #include <d3d11_4.h>
+        #include <codecvt>
+        #include "opencv2/core/directx.hpp"
+        #ifdef HAVE_OPENCL
+            #include <CL/cl_d3d11.h>
+        #endif
+
+namespace cv {
+namespace gapi {
+namespace wip {
+VPLDX11AccelerationPolicy::VPLDX11AccelerationPolicy()
+{
+#ifdef CPU_ACCEL_ADAPTER
+    adapter.reset(new VPLCPUAccelerationPolicy);
+#endif
+}
+
+VPLDX11AccelerationPolicy::~VPLDX11AccelerationPolicy()
+{
+    if (hw_handle)
+    {
+        GAPI_LOG_INFO(nullptr, "VPLDX11AccelerationPolicy release ID3D11Device");
+        hw_handle->Release();
+    }
+}
+
+void VPLDX11AccelerationPolicy::init(session_t session) {
+    mfxStatus sts = MFXVideoCORE_GetHandle(session, MFX_HANDLE_D3D11_DEVICE, reinterpret_cast<mfxHDL*>(&hw_handle));
+    if (sts != MFX_ERR_NONE)
```

Comment:
I'd recommend using `GAPI_Assert`

Reply:
i do not want to use assert mostly in the code because DX11 uses COM (Common Object Model) IUnknown interface with AddRef & Release semantic and i concerned about proper deinitialization (Will introduce CComPtr further). Moreover, I'll expect  some kind of bypass behavior - when DX11 in unavailble but implementations falls to CPU accel